docs: specify what happens when there are rules with the same name in an extended config (#5012)
Mergifyio/docs: src/content/docs/configuration/sharing.mdx

  - Values in the `default` key will be merged and remote default values will
    apply to local configuration.

  - If there is rule in any of the `pull_request_rules`, `queue_rules`, `priority_rules`, `partition_rules`
    or `merge_protections` section with the same name in both the parent and child config, then the duplicated

>     rule in the child config will NOT be overwritten with rule from the parent config.

Reviewer: child/parent wording is confusing here, I have no idea if the child is the one from `extends` or not.

Author: Modified it to describe which is which

  - Values in the `default` key will be merged and remote default values will
    apply to local configuration.

>   - If there is rule in any of the `pull_request_rules`, `queue_rules`, `priority_rules`, `partition_rules`

Reviewer 2: ```suggestion
  - If there is a rule in any of the `pull_request_rules`, `queue_rules`, `priority_rules`, `partition_rules`
```

>   - If there is a rule in any of the `pull_request_rules`, `queue_rules`, `priority_rules`, `partition_rules`

Reviewer 3: ```suggestion
  - If there is a rule in any of the `pull_request_rules`, `queue_rules`, `priority_rules`, `partition_rules`,
```

Author: Since there's a `or` after this word, i don't think a comma is here appropriate @DouglasBlackwood

Reviewer 3: True in French, not in English 😄

Author: I was intrigued so i searched up a bit, and looks like both are valid syntax IIUC: https://www.grammarly.com/blog/comma-before-or/ @DouglasBlackwood

Author: Added the comma

Reviewer 3: Interesting, thanks for the link.

>     or `merge_protections` section with the same name in both the extended config and the

Reviewer 4: what do you think of introducing the words `parent config` and `child config` to make the concept easier to grasp?

Author: Good idea, made a modification, let me know if that sounds good know

>   - If there are `pull_request_rules`, `queue_rules`, `priority_rules`, `partition_rules`
>     or `merge_protections` rules with the same name in both the extended config and the
>     config from which it is extended, then the rule, with the same name, in the config that
>     extends the other will NOT be overwritten.

Reviewer 2: ```suggestion
  - If there are `pull_request_rules`, `queue_rules`, `priority_rules`, `partition_rules`
    or `merge_protections` rules with the same name in both the extended config and the
    config from which it is extended, then the duplicated rule will NOT be overwritten. Thus, the rule
    defined in the config containing the `extends` keyword will be used.
```
I had some troubles understanding, may be we can turn it in another way ?

My suggestion is just an example we can change it to be clearer, wdyt ?
